state/factory/workingset.go

```diff
@@ -752,7 +818,9 @@ func (ws *workingSet) ValidateBlock(ctx context.Context, blk *block.Block) error
 			return err
 		}
 	}
-	if err := ws.processWithCorrectOrder(ctx, blk.RunnableActions().Actions()); err != nil {
+	var err error
+	err = ws.Process(ctx, blk.RunnableActions().Actions())
+	if err != nil {
```

Comment:
so `Process` needs to use legacy order before the hardfork, while `Validate()` can always use correct order? or is it vice-versa?

Reply:
`if err := ws.Process(); err != nil`